### PR TITLE
Add softmin op

### PR DESCRIPTION
**Pull request by beverlylytle**
Adds support for the `softmin` op, which is nearly identical to the `softmax` op.  However, unlike for `softmax` there is no `torch.softmin`  nor `torch.Tensor.softmin`.  

Also, it looks like `softmax` is supported on CPU for float16 now.

### CODE REVIEW

File: thunder/tests/opinfos.py [hunk 1]
```diff
@@ -8161,38 +8161,34 @@ def softmax_sample_generator(op, device, dtype, requires_grad, **kwargs):
     sample_input_generator=softmax_sample_generator,
     torch_reference=torch.softmax,
     dtypes=(datatypes.floating,),
-    test_directives=(
```
Comment (mruberry) [inline thunder/tests/opinfos.py:8164]: Awesome!

**Review by mruberry — APPROVED** [review]
